src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1395,7 +1395,12 @@ export class BundleDataClient {
           // will usually be called in production with block ranges that were validated by
           // DataworkerUtils.blockRangesAreInvalidForSpokeClients.
           const startBlockForChain = Math.min(_startBlockForChain, spokePoolClient.latestBlockSearched);
-          const endBlockForChain = Math.min(_endBlockForChain, spokePoolClient.latestBlockSearched);
+          // @dev Add 1 to the bundle end block. The thinking here is that there can be a gap between
+          // block timestamps in subsequent blocks. The bundle data client assumes that fill deadlines expire
+          // in exactly one bundle, therefore we must make sure that the bundle block timestamp for one bundle's
+          // end block is exactly equal to the bundle block timestamp for the next bundle's start block. This way
+          // there are no gaps in block timestamps between bundles.
+          const endBlockForChain = Math.min(_endBlockForChain + 1, spokePoolClient.latestBlockSearched);
```

Comment:
The end timestamp of a bundle now overlaps with the start block of the next bundle. Maybe it is just safer to do this in addition with `endTime = await spokePoolClient.getTimestampForBlock(endBlockForChain) - 1`?

Reply:
Hmm well I think that's ok because of the way we check bundle block timestamps [here](https://github.com/across-protocol/sdk/blob/57ce0138c77721ccde3c65ad3be99b13fe69e295/src/clients/BundleDataClient/BundleDataClient.ts#L847) and [here](https://github.com/across-protocol/sdk/blob/57ce0138c77721ccde3c65ad3be99b13fe69e295/src/clients/BundleDataClient/BundleDataClient.ts#L1051) for example.

But I think you're right. By subtracting 1 from it, we'll be strictly safer and also guarantee no gaps

Reply:
6e90289edb73ac4ce62c3fe26006171417527a36